src/protocol.h

```diff
@@ -136,25 +136,42 @@ class CInv
 public:
     CInv();
     CInv(int typeIn, const uint256& hashIn);
+    CInv(int typeIn, const uint256& hashIn, const uint256& hashAuxIn);
 
     ADD_SERIALIZE_METHODS;
 
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action)
     {
+        int nVersion = s.GetVersion();
         READWRITE(type);
         READWRITE(hash);
+        if (type == MSG_WTX && nVersion >= CINV_WTX_VERSION) {
+            READWRITE(hashAux);
+        } else if (type == MSG_TX && ser_action.ForRead()) {
+            // Ensure that this value is set consistently in memory for MSG_TX.
+            hashAux = uint256S("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
```

Comment:
Please extract as a named constant, counting `f`s is hard. I've counted 3 times and come up with the right number once. :)